service/header/core_listener.go

```diff
@@ -38,93 +38,59 @@ func NewCoreListener(p2pSub *P2PSubscriber, fetcher *core.BlockFetcher, dag form
 
 // Start kicks off the CoreListener listener loop.
 func (cl *CoreListener) Start(ctx context.Context) error {
-	cl.ctx, cl.cancel = context.WithCancel(context.Background())
+	if cl.blockSub != nil {
+		return fmt.Errorf("core-listener: already started")
+	}
```

Comment:
The whole purpose of setting some value(either ctx with cancel or blockSub like I did here) to nil in `Stop` is to check that `Start` wouldn't start things twice, so this if is needed. This is not required, like setting values to nil on  `Stop`, but if we want to have this runtime restart ability, both are needed.